space-plugins/nuxt-base/server/api/_oauth.post.ts

```diff
@@ -0,0 +1,37 @@
+export default defineEventHandler(async (event): Promise<OAuthResponse> => {
+	const appConfig = useAppConfig();
+	const { initOAuth } = await readBody(event);
+	if (initOAuth) {
+		setCookie(event, AUTH_COOKIE_NAME, '', {
```

Comment:
Is there a specific reason to set the cookie here ? I thought I would be set by the app-extension auth once it gets trough the flow.

Reply:
this is to **_clear_** the stale cookie in case a user re-installed the extension and needs to restart the whole oauth cookie. (do you remember the issue with the redirection the iframe)

Reply:
Oh yeah that makes sense. The question that I am asking myself/where I am torn, is if it makes more sense to somehow to handle this cookie reset inside app-exertion auth or in the backend, so that there are not many pieces floating around. Because when we decide to switch from the cookie approach and decide to store the cookie in a DB this behavior is loosely coupled with the rest so it might be harder to think of and consider then.

Reply:
Oh I see what you mean. Yeah it makes sense. Let's do the work on app-extension-auth to allow adaptors, and then move this logic back to app-extension-auth.